src/main/java/com/ocient/jdbc/XGConnection.java

```diff
@@ -3073,7 +3104,7 @@ private String getSchemaFromServer() throws Exception
 			}
 			else
 			{
-				reconnect();
+				blockingGetSQLResult(reconnect());
```

Comment:
I don't think we should swallow this exception
```suggestion
				blockingGetSQLResult(reconnect()).get();
```